torch_geometric/data/temporal.py

```diff
@@ -117,11 +116,12 @@ def __getitem__(self, idx: Any) -> Any:
 
         prepared_idx = self.__prepare_non_str_idx(idx)
 
-        data = copy.copy(self)
-        for key, item in data:
-            if item.size(0) == self.num_events:
+        data = {}
```

Comment:
What's the reason for this change?

Reply:
When using the default implementation of the `DataLoader` with Temporal Data, one bottleneck is this method. With the new implementation, it is around 10% more performative. I can revert this change if you think it is not necessary.

Reply:
Interesting. Any idea what slows down the code in the existing implementation?

Reply:
It seems to be the `copy` of the entire `temporalData` object when getting an item.

Reply:
Reverted back to `copy` since I think it is important to maintain any attributes with `item.size(0) != self.num_events`.